### PR TITLE
adminguide: add security section

### PR DESCRIPTION
**Pull request by garlick**
Problem: system administrators need to be aware of how Flux security works.

Add an overview subsection on Flux security to the admin guide.

### CODE REVIEW

File: adminguide.rst [hunk 1]
```diff
@@ -162,6 +162,71 @@ FLUID
...
+via a UNIX domain socket at a well known location, usually ``/run/flux/local``.
+Users are authenticated on this socket using the SO_PEERCRED socket option.
+Once connected, a user may inject messages into the overlay network.  Messages
+are stamped by the broker at ingress with the user's authenticated userid,
+and a *role mask* that identifies any special capabilities granted to the user.
+Messages that are sent by the ``flux`` user are stamped with the instance owner
+role, while other users, or *guests*, are stamped with a role that grants
+minimal access.  Note that the ``root`` user is considered a guest user with
+no special privilege in Flux, but sites can choose to grant ``root`` the owner
+role by configuration if desired.
```
Comment (grondo) [inline adminguide.rst:192]: Suggestion: add
```
See :core:man5:`flux-config-security`
```

or similar here?

File: adminguide.rst [hunk 2]
```diff
@@ -162,6 +162,71 @@ FLUID
...
+drain request messages that have the owner role.  Similarly, any job can be
+canceled by a cancel request message with the owner role, but in addition,
+jobs can be canceled by non-owners whose message userid matches the target
+job userid.
+
+A Flux job is launched when brokers launch one :core:man1:`flux-shell` per
+node with the credentials of the user that submitted the job.  When that is a
+guest user, Flux employs a setuid helper called the :security:man8:`flux-imp`
+to launch the shells with the guest credentials.  The shells in turn launch
+one or more user processes that comprise the parallel job.
```
Comment (grondo) [inline adminguide.rst:206]:
```suggestion
one or more user processes that compose the parallel job.
```

I _think_ the job _comprises_ one or more user processes, but the processes _compose_ the job.

**Review by grondo — APPROVED** [review]
Awesome! Very well written and critically informative. Thanks!

Just a couple minor suggestions inline.

File: adminguide.rst (hunk 2 above)
Comment (garlick) [reply adminguide.rst:206]: I think you're right!

**Comment by garlick** [comment]
Thanks - I'll set MWP

**Pull request merged**